src/middlewared/middlewared/alert/source/active_directory.py

```diff
@@ -20,6 +20,13 @@ class ActiveDirectoryDomainHealthAlertClass(AlertClass):
     text = "Domain validation failed with error: %(verrs)s."
 
 
+class ActiveDirectoryDomainOfflineAlertClass(AlertClass, SimpleOneShotAlertClass):
```

Comment:
Who creates this alert?

Reply:
Alert is generated when domain state transitions to offline. Is deleted when it comes back online. see parse_msg().